hazelcast/src/main/java/com/hazelcast/wan/ReplicationEventObject.java

```diff
@@ -16,8 +16,21 @@
 
 package com.hazelcast.wan;
 
+import com.hazelcast.nio.serialization.Data;
+
 /**
- * Marker interface for WAN replication messages
+ * Interface for all WAN replication messages
  */
 public interface ReplicationEventObject {
```

Comment:
can we move this interface to `impl` package? or do we expect users to implement this?

Reply:
No, users do not implement it. I'll move it.

Reply:
Seems we can't move it since it's a method argument in `WanReplicationPublisher` which can be implemented by the user. Moving it would cause existing implementations of `WanReplicationPublisher` to not compile.